open_flamingo/eval/evaluate.py

```diff
@@ -442,7 +497,9 @@ def get_prompt(sample):
 
     metrics = compute_cider(
         result_path=results_path,
-        annotations_path=annotations_json_path,
+        annotations_path=args.coco_annotations_json_path
+        if dataset_name == "coco"
```

Comment:
the line `if dataset_name == "coco"` appears so many times, at the very least it should be a variable, but I wonder whether there is a simpler way to refactor the code so these if/else blocks aren't used all over the place